flocker/node/agents/functional/test_cinder.py

```diff
@@ -107,3 +107,8 @@ def test_foreign_cluster_volume(self):
         self.addCleanup(blockdevice_api2.destroy_volume,
                         flocker_volume.blockdevice_id)
         self.assert_foreign_volume(flocker_volume)
+
+    # Maybe add an implementation specific test here to verify that
+    # the ID returned by compute_instance_id() can be retreived using
+    # the nova server get API and that it has our IP addesses
+    # http://developer.openstack.org/api-ref-compute-v2.html#getServer
```

Comment:
I think our existing test for `get_device_path` might cover this.  It requires that after attaching to our `compute_instance_id` the reported device path is an actual block device on the system where the tests are running.  That's a pretty good indication that we've gotten the right compute instance id.